Tasks/AzureRmWebAppDeployment/AzureRmWebAppDeployment.ps1

```diff
@@ -1,134 +1,142 @@
-param
-(
-    [String] [Parameter(Mandatory = $true)]
-    $ConnectedServiceName,
-
-    [String] [Parameter(Mandatory = $true)]
-    $WebAppName,
+[CmdletBinding()]
+param()
+
+Trace-VstsEnteringInvocation $MyInvocation
+
+try{
+
+	# Get inputs.
+	$WebAppName = Get-VstsInput -Name WebAppName -Require
+	$DeployToSlotFlag = Get-VstsInput -Name DeployToSlotFlag -Require
+	$ResourceGroupName = Get-VstsInput -Name ResourceGroupName
+	$SlotName = Get-VstsInput -Name SlotName
+	$Package = Get-VstsInput -Name Package -Require
+	$RemoveAdditionalFilesFlag = Get-VstsInput -Name ScriptArgument
+	$ExcludeFilesFromAppDataFlag = Get-VstsInput -Name ExcludeFilesFromAppDataFlag 
+	$TakeAppOfflineFlag = Get-VstsInput -Name TakeAppOfflineFlag 
+	$VirtualApplication = Get-VstsInput -Name VirtualApplication 
+	$AdditionalArguments = Get-VstsInput -Name AdditionalArguments 
+	$WebAppUri = Get-VstsInput -Name WebAppUri
+	$SetParametersFile = Get-VstsInput -Name SetParametersFile
+
+	# Initialize Azure.
+
+	Import-Module $PSScriptRoot\ps_modules\VstsAzureHelpers_
+	Initialize-Azure
+
+	# Import the loc strings.
+	Import-VstsLocStrings -LiteralPath $PSScriptRoot/Task.json
+
+
+	Write-Verbose "Starting AzureRM WebApp Deployment Task"
+
+	Write-Verbose "ConnectedServiceName = $ConnectedServiceName"
+	Write-Verbose "WebAppName = $WebAppName"
+	Write-Verbose "DeployToSlotFlag = $DeployToSlotFlag"
+	Write-Verbose "ResourceGroupName = $ResourceGroupName"
+	Write-Verbose "SlotName = $SlotName"
+	Write-Verbose "Package = $Package"
+	Write-Verbose "SetParametersFile = $SetParametersFile"
+	Write-Verbose "RemoveAdditionalFilesFlag = $RemoveAdditionalFilesFlag"
+	Write-Verbose "ExcludeFilesFromAppDataFlag = $ExcludeFilesFromAppDataFlag"
+	Write-Verbose "TakeAppOfflineFlag = $TakeAppOfflineFlag"
+	Write-Verbose "VirtualApplication = $VirtualApplication"
+	Write-Verbose "AdditionalArguments = $AdditionalArguments"
+	Write-Verbose "WebAppUri = $WebAppUri"
+
+	$WebAppUri = $WebAppUri.Trim()
+	$Package = $Package.Trim('"').Trim()
+
+	if( [string]::IsNullOrEmpty($Package) ){
+		Throw (Get-VstsLocString -Key "Invalidwebapppackagepathprovided")
+	}
 
-    [String] [Parameter(Mandatory = $true)]
-    $DeployToSlotFlag,
+	if ($ExcludeFilesFromAppDataFlag -match '[\r\n]' -or [string]::IsNullOrEmpty($ExcludeFilesFromAppDataFlag)) {
```

Comment:
This looks like a bug that you're matching against carriage return or newline to determine whether to default to false. I think you just need to use the `-AsBool` switch when you call `Get-VstsInput` and nothing else required here.